### PR TITLE
qol(islands): improve hydration errors

### PR DESCRIPTION
**Pull request by lilnasy**
## Changes

- https://github.com/withastro/astro/issues/10055
- Improves error messages for when hydration fails

<details>
<summary>Before</summary>

image

</details>

<details>
<summary>After</summary>

image

</details>

## Testing
Does not affect behavior.

## Docs
Does not affect usage.

### CODE REVIEW

**Comment by lilnasy** [comment]
!preview island-loading-error-messages

File: packages/astro/src/runtime/server/astro-island.ts [hunk 1]
```diff
@@ -93,14 +93,15 @@ declare const Astro: {
...
+				async start() {
 					const opts = JSON.parse(this.getAttribute('opts')!) as Record<string, any>;
 					const directive = this.getAttribute('client') as directiveAstroKeys;
 					if (Astro[directive] === undefined) {
 						window.addEventListener(`astro:${directive}`, () => this.start(), { once: true });
 						return;
 					}
-					Astro[directive]!(
+					try {
+					await Astro[directive]!(
```
Comment (ematipico) [inline packages/astro/src/runtime/server/astro-island.ts:104]: Is this change safe?

**Review by ematipico — APPROVED** [review]

File: packages/astro/src/runtime/server/astro-island.ts (hunk 1 above)
Comment (lilnasy) [reply packages/astro/src/runtime/server/astro-island.ts:104]: What could be an issue?

Comment (ematipico) [reply packages/astro/src/runtime/server/astro-island.ts:104]: I'm genuinely asking 😂 sometimes we don't wait for promises on purpose

Comment (lilnasy) [reply packages/astro/src/runtime/server/astro-island.ts:104]: The one thing I can think of is a "thenable". But if a custom client directive is using it in a fancy way, it is just asking for trouble.

```js
await { then(resolve) { resolve(5) } }
```

Comment (ematipico) [reply packages/astro/src/runtime/server/astro-island.ts:104]: Thank you!

**Review by natemoo-re — APPROVED** [review]

**Pull request merged**